### PR TITLE
Etabs toolkit issue140 control auto offset rigid zone factor

### LINKED ISSUE #140: Etabs_Toolkit: Control auto offset RigidZone factor

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

 Closes #140 

 
Adding possibility to control RigidZone offset for the Autolength offset.
Adding AutoLengthOffset object in the Etabs_oM to control this.
Changing the previous setter, just setting a boolean, to setting this object as well, with a default RZ-factor of 1.0

 ### Test files

https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/ETABS_Toolkit/Etabs_Toolkit-Issue140-RigidZoneFactor?csf=1&e=rcgk4W

### CODE REVIEW

**Comment by JosefTaylor** [comment]
Compiles, and rigid zone factor works as promised, but autoLengthOffset only sets the rigid zone factor to 0, rather than changing the AutoOffset boolean.

File: Etabs_Adapter/CRUD/Create.cs [hunk 1]
```diff
@@ -204,7 +204,10 @@ private bool CreateObject(Bar bhBar)
...
             if (bhBar.AutoLengthOffset())
             {
-                if (m_model.FrameObj.SetEndLengthOffset(name, true, 0, 0, 0) != 0)
+                //the Rigid Zone Factor is not picked up when setting the auto length = true for the method call. Hence need to call this method twice.
+                int retAutoLEngthOffset = m_model.FrameObj.SetEndLengthOffset(name, false, 0, 0, bhBar.AutoLengthOffsetRigidZoneFactor());
+                retAutoLEngthOffset += m_model.FrameObj.SetEndLengthOffset(name, true, 0, 0, 0);
+                if (retAutoLEngthOffset != 0)
                 {
                     CreatePropertyWarning("Auto length offset", "Bar", name);
                     ret++;
```
Comment (JosefTaylor) [inline Etabs_Adapter/CRUD/Create.cs]:
```suggestion
            else
            {
                if (m_model.FrameObj.SetEndLengthOffset(name, false, 0, 0, 0, ETABS2016.eItemType.Objects) != 0)
                {
                    CreatePropertyWarning("Length offset", "Bar", name);
                    ret++;
                }
            }
```

**Comment by IsakNaslundBh** [comment]
Ah, think I see what you mean. So will need to be a triple if else then, if AutoOffset, else if offset != null, else your suggestion.

Will fix

**Comment by IsakNaslundBh** [comment]
Updated the strategy used for this to only set the auto length offset if it had been defined explicitly, otherwise, do nothing and let Etabs use the default settings (which sets auto length offset with rigid zone factor of 0).

**Comment by JosefTaylor** [comment]
This looks great. Compiles and behavior of setting AutoOffset to False is as expected.

**Pull request merged**